test/OrchardCore.Commerce.Tests.UI/Tests/TaxTests/TaxBehaviorTests.cs

```diff
@@ -130,15 +131,23 @@ public Task CustomTaxRateEditorShouldWork(Browser browser) =>
                 void ResetScroll() => context.ExecuteScript("window.scrollTo(0, 0);");
 
                 By ByCell(int index, string name) =>
-                    By.Name($"ISite.Rates[{index.ToTechnicalString()}].{name}");
+                    By.CssSelector($".taxRateSettings__row_{index.ToTechnicalString()} .taxRateSettings__{name.ToCamelCase()}");
 
                 Task SetCellAsync(int index, string name, string value) =>
                     context.ClickAndFillInWithRetriesAsync(ByCell(index, name), value);
 
                 Task SetCorporationAsync(int index, MatchTaxRates value) =>
                     context.SetDropdownByValueAsync(
                         ByCell(index, nameof(TaxRateSetting.IsCorporation)),
-                        ((int)value).ToTechnicalString());
+                        value.ToString());
+
+                void SetTaxRate(int index, decimal value)
+                {
+                    var element = context.Get(ByCell(index, nameof(TaxRateSetting.TaxRate)));
+                    element.Click();
+                    element.Clear();
+                    element.SendKeysWithLogging(value.ToTechnicalString());
```

Comment:
Why not `ClickAndFillInWithRetriesAsync`? It does click-clear-send keys with logging, but also with retries and more logging.

Reply:
That's very strange. It should just set the.l literal string.

The retries are useful in case the text entry doesn't actually happen, what's a random occurrence.